mongodb/resource_automation_agent.go

```diff
@@ -0,0 +1,127 @@
+	data.SetId(conn.ToJSON())
+
+	// read process config
+	automation := data.Get("automation").([]interface{})
+	automationConfig := types.ReadAutomationAgentConfig(automation)
+
+	// create a SSH connection to the remote host
+	client, err := NewSSHClient(providerConfig, conn)
+	if err != nil {
+		return fmt.Errorf("could not create a SSH client: %v", err)
+	}
+
+	// attempt to create directories if not already present
+	cmd := fmt.Sprintf("mkdir -p %s %s", automationConfig.AgentDir, automationConfig.LogPath)
+	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
+
+	// Set correct permissions for directories
+	cmd = fmt.Sprintf("chown `whoami` %s %s", automationConfig.AgentDir, automationConfig.LogPath)
+	ssh.PanicOnError(client.RunCommand(conn.SudoPrefix(cmd)))
+
+	// download the automation agent binary on the local host
+	localFile, err := util.DownloadFile(automationConfig.Binary)
+	util.PanicOnNonNilErr(err)
+	defer util.LogError(localFile.Close)
+	log.Printf("[DEBUG] downloaded automation agent binary to: %s", localFile.Name())
+
+	// upload the binary
+	remoteFilePath := path.Join(automationConfig.AgentDir, filepath.Base(localFile.Name()))
+	ssh.PanicOnError(client.UploadFile(remoteFilePath, localFile))
+
+	// unpack the binary
+	cmd = fmt.Sprintf("tar -C %s -xvzf %s --strip 1", automationConfig.AgentDir, remoteFilePath)
+	ssh.PanicOnError(client.RunCommand(cmd))
+	log.Printf("[DEBUG] unpacked the binary in: %s", automationConfig.AgentDir)
+
+	// TODO: get the API key and projectID from client
+	// modify automation agent config: baseUrl, ApiKey, and projectID must be set in the file along with any specified overrides
+	err =
+		updatePropertiesFile(client, conn, automationConfig.ConfigFilename(), func(props *opsmanagerconfig.PropertiesFile) {
+			props.SetPropertyValue(automationConfig.GetAutomationConfigTag("ProjectID"), automationConfig.ProjectID)
+			props.SetComments(automationConfig.GetAutomationConfigTag("ProjectID"), []string{"", commentString, ""})
+
+			props.SetPropertyValue(automationConfig.GetAutomationConfigTag("APIKey"), automationConfig.APIKey)
+			props.SetComments(automationConfig.GetAutomationConfigTag("APIKey"), []string{"", commentString})
+
+			props.SetPropertyValue(automationConfig.GetAutomationConfigTag("BaseURL"), automationConfig.BaseURL)
+			props.SetComments(automationConfig.GetAutomationConfigTag("BaseURL"), []string{"", commentString})
+			for prop, val := range automationConfig.Overrides {
+				props.SetPropertyValue(prop, val.(string))
+			}
+		})
+	util.PanicOnNonNilErr(err)
+
+	// start the automation agent
+	cmd = fmt.Sprintf("nohup ./mongodb-mms-automation-agent --config=%s >> %s/automation-agent-fatal.log 2>&1 &", automationConfig.ConfigFilename(), automationConfig.LogPath)
+	ssh.PanicOnError(client.RunCommand(cmd))
+	log.Printf("[DEBUG] started automation agent from configuration file %s", automationConfig.ConfigFilename())
+
+	// TODO check if it's running
```

Comment:
something like `(pgrep mongodb-mms-automation-agent && echo "Started" ) || echo "Stopped"` should work